### PR TITLE
Move Helpsystem to module

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Decouple Help System from SMA and move 

## PR Context

Related https://github.com/PowerShell/PowerShell/issues/3954#issuecomment-499389006

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by iSazonov** [comment]
I reverted last commit moving HelpDisplayStrings.resx from SMA to new module - I don't know how resolve this problem.

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 80 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

TestGetContentWriter
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentWriter() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 175
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestClearProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearProperty() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 152
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestGetContentReader
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentReader() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 162
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestClearContent
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearContent() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 189
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestSetProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestSetProperty() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 137
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```

**Comment by PoshChan** [comment]
@iSazonov, your last commit had 20 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

TestGetContentWriter
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentWriter() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 175
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestClearProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearProperty() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 152
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestGetContentReader
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentReader() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 162
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestClearContent
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearContent() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 189
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestSetProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestSetProperty() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 137
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs [hunk 1]
```diff
@@ -64,6 +64,7 @@ public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray
                 System.Diagnostics.Debugger.Break();
```
Comment (PoshChan) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: @TravisEz13, your last commit had 20 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

TestGetContentWriter
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentWriter() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 175
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestClearProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearProperty() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 152
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestGetContentReader
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentReader() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 162
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestClearContent
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearContent() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 189
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestSetProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestSetProperty() in /Users/vsts/agent/2.153.2/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 137
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /Users/vsts/agent/2.153.2/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: @TravisEz13, your last commit had 20 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

TestGetContentWriter
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentWriter() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 175
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestClearProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearProperty() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 152
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestGetContentReader
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentReader() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 162
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestClearContent
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearContent() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 189
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```
TestSetProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 1589
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestSetProperty() in /home/vsts/work/1/s/test/xUnit/csharp/test_FileSystemProvider.cs:line 137
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in /home/vsts/work/1/s/src/System.Management.Automation/engine/InitialSessionState.cs:line 4550
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs]: @TravisEz13, your last commit had 81 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

TestGetContentWriter
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentWriter() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 175
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestClearProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearProperty() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 152
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestGetContentReader
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestGetContentReader() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 162
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestClearContent
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestClearContent() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 189
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```
TestSetProperty
```powershell
System.TypeInitializationException : The type initializer for 'System.Management.Automation.Runspaces.InitialSessionState' threw an exception.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Management.Automation.Runspaces.InitialSessionState.CreateDefault2() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 1587
   at PSTests.Parallel.FileSystemProviderTests.GetExecutionContext() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 51
   at PSTests.Parallel.FileSystemProviderTests.GetProvider() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 59
   at PSTests.Parallel.FileSystemProviderTests.TestSetProperty() in D:\a\1\s\test\xUnit\csharp\test_FileSystemProvider.cs:line 137
----- Inner Stack Trace -----
   at System.Management.Automation.Runspaces.InitialSessionState..cctor() in D:\a\1\s\src\System.Management.Automation\engine\InitialSessionState.cs:line 4550
```

**Comment by iSazonov** [comment]
@adityapatwardhan I was blocked because there are a breaking changes, depth engine dependencies and  new public API is needed. I asked @SteveL-MSFT and he suggested to pull the draft so that you could  help move forward or reject. I guess it is impossible to rebase the code but we could discuss how we could decouple the module if you and team have time and interest.

**Pull request closed without merging**